enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1158 @@
+Hierarchy](#change-management-hierarchy) for caveats).
+
+All change management strategies, except `Disabled`, are subject to the following `changeManagement` fields:
+- `changeManagement.disabledUntil: <bool|date>`: When `disabledUntil: true` or `disabledUntil: <future-date>`, the interpreted strategy for 
+  change management in the resource is `Disabled`. Setting a future date in `disabledUntil` offers a less invasive (i.e. no important configuration needs to be changed) method to 
+  disable change management constraints (e.g. if it is critical to roll out a fix) and a method that
+  does not need to be reverted (i.e. it will naturally expire after the specified date and the configured
+  change management strategy will re-activate).
+- `changeManagement.pausedUntil: <bool|date>`: Unless the effective active strategy is Disabled, `pausedUntil: true` or `pausedUntil: <future-date>`, change management must 
+  pause material changes.
+
+### Change Management Status
+Change Management information will also be reflected in resource status. Each resource 
+which contains the stanza in its `spec` will expose its current impact in its `status`. 
+Common user interfaces for aggregating and displaying progress of these underlying resources
+should be updated to proxy that status information to the end users.
+
+### Change Management Metrics
+Cluster wide change management information will be made available through cluster metrics. Each resource
+containing the stanza should expose the following metrics:
+- The number of seconds until the next known permitted change window. 0 if changes can currently be initiated. -1 if changes are paused indefinitely. -2 if no permitted window can be computed.
+- Whether any change management strategy is enabled.
+- Which change management strategy is enabled.
+- If changes are pending due to change management controls.
+
+### Change Management Hierarchy
+Material changes to worker-nodes are constrained by change management policies in their associated resource AND 
+at the control-plane resource. For example, in a standalone profile, if a MachineConfigPool's change management
+configuration apparently permits material changes from being initiated at a given moment, that is only the case 
+if ClusterVersion is **also** permitting changes from being initiated at that time.
+
+The design choice is informed by a thought experiment: As a cluster lifecycle administrator for a Standalone cluster,
+who wants to achieve the simple goal of ensuring no material changes take place outside a well-defined 
+maintenance schedule, do you want to have to the challenge of keeping every MachineConfigPool's 
+`changeManagement` stanza in perfect synchronization with the ClusterVersion's? What if a new MCP is created 
+without your knowledge?
+ 
+The hierarchical approach allows a single master change management policy to be in place across 
+both the control-plane and worker-nodes. 
```

Comment:
> It still feels a bit cumbersome to add the same scheduling info to several different resources.
It may already be clear, but just in case: the hierarchy approach is so that adding the same info to multiple resources _can_ be avoided. If an operations team for a standalone cluster wants a simple maintenance schedule, they add it to `ClusterVersion`. This controls changes to the control-plane AND worker-nodes.
If they want more complex approach, they add more specific maintenance schedule for a MCP1 and a different one for MCP2. 

If you instead mean, why can't the schedule information for all components be in the same CRD, that was an [early approach I documented](https://github.com/jupierce/oc-update-design-ideas/commit/a6364ee2f2c1ebf84ed6d50bc277f9736bf793bd) for the working group. Ultimately, it felt less intuitive across profiles. Consider a CRD called `UpdatePolicy` that tries to be a central config.
1. Is it namespaced? If it is, that feels odd for an object that that should control the cluster. If it is not namespaced, the policy feels misplaced for HCP HostedCluster resources which live in a namespace. 
2. Where does someone check the status of the policy (e.g. when the next update is going to be possible on a given MCP?). If it is the `UpdatePolicy` object, you have multiple independent controllers controlling the status, which is the start of a bad time. If the UpdatePolicy controls multiple different MCPs differently, how do they independently report their status'? It also introduces the problem of having to look at multiple places (MCP and `UpdatePolicy` to understand what may be happening.
3. As you pack policies for control-plane, MCPs, HCP primitives into an expressive `UpdatePolicy`, the schema options varied from too complex, to error prone, to overly abstract, to overly limiting. 
4. If you split the policy object up to simplify it, e.g. one for each MCP, you have the problem of associating it to the right MCP unambiguously. Planting the policy in the MCP itself solves this problem.

In summary, it possible, but I didn't consider what I came up with more elegant or user friendly than housing the policies in the resources they control.

> Alternately, what if we only had the enabled/disabled flag and left it up to an outside actor (ACM, OCM, etc.) to implement schedules? That would also open up use cases like "allow updates on Saturdays, unless service X is very busy serving traffic" via user scripting rather than requiring us to build the feature in ourselves.
I seek to expose those primitives (see `disabledUntil`, Manual, or Assisted) to allow customers to construct arbitrarily complex or simple methods for choreographing updates. 
`MaintenanceSchedule` is the OOB strategy that should serve the 80%.

Reply:
> In summary, it possible, but I didn't consider what I came up with more elegant or user friendly than housing the policies in the resources they control.

That all makes sense, thanks for the detailed response.

Reply:
Added this discussion to the Alternative section.